src/Core/Session.php

```diff
@@ -278,4 +170,75 @@ public static function isAuthenticated()
 
 		return $_SESSION['authenticated'];
 	}
+
+	/**
+	 * @brief Calculate the hash that is needed for the "Friendica" cookie
+	 *
+	 * @param array $user Record from "user" table
+	 *
+	 * @return string Hashed data
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	private static function getCookieHashForUser($user)
```

Comment:
Shouldn't this be moved to `Model\User`?

Reply:
I moved it to a own class `Model\User\Cookie` so we don't create "monster" classes :-)